Fix issue with different `caught?` exception msg in Clojure 1.11

ikappaki opened the pull request:
Hi,

could you please consider patch to fix issue with `caught?` on Clojure 1.11. Fixes #4 

Basically, `caught?` inspects the exception thrown to see if it matches `symbol`. Symbol in versions up to 1.10 was a FQ var, with Clojure 1.11 is a FQ symbol (it does not contain the leading `#'`). The exception msg matching code has been updated to ignore `#'`.

I've also introduced a GH CI action to run the CLJ, CLJS tests for clojure 1.10 and 1.11, I hope you don't mind. Not sure if there is a better way to specify the clojure version to use in `deps.edn`

Thanks

ikappaki commented:
Oops I've missed the .circleci tests! I though they were none! Looking into them now.

borkdude commented:
I'll take a look tomorrow to fix the build :)

ikappaki commented:
> I'll take a look tomorrow to fix the build :)

I'll try to convert the scripts into bb tasks if you don't mind and update the `circleci` config. I've only introduced a GH action config because I though there were no CI tests, I'll remove that.

borkdude commented:
I think the circleci image may need updating

ikappaki commented:
> I think the circleci image may need updating

I'll have a look, thanks

ikappaki commented:
> > I think the circleci image may need updating
> 
> I'll have a look, thanks

I've managed to convert the scripts to bb tasks and update the circleci image/configuration (haven't pushed a commit yet). All tests pass except I can't compile the clojurescript tests for node with anything above clojurescript 1.10.597 (the next version [1.10.741](https://github.com/clojure/clojurescript/blob/master/changes.md) has the Rhino/Nashorn/Graalvm repls removed, could this be relevant?). I'll continue the investigation tomorrow ...

borkdude commented:
hi @ikappaki - I updated and simplified the build (dropped planck due to problems with some native lib, we already have node tests which seem sufficient).

ikappaki commented:
> hi @ikappaki - I updated and simplified the build (dropped planck due to problems with some native lib, we already have node tests which seem sufficient).

Hi @borkdude, ah I am about to make a commit with a next gen image while still keeping planck tests in place, shall I go ahead and demo this with my soon to commit patch? we can always revert back to your update of course during the review.

borkdude commented:
I'm ok with dropping planck as this is a recurring issue.

ikappaki commented:
Hi @borkdude,

I have committed a follow up patch with the following changes.

1. Explicitly sets the `cljs.main` output dir to `out` when compiling the tests, this is a workaround for what seems to be a regression I've just logged at https://ask.clojure.org/index.php/12335/clojurescript-cljs-main-compilation-recent-releases-create
2. Refactor the clojure version aliases in `deps.edn` introduced in the previous commit to have better names and share a common base.
3. Converted the `script`s to bb tasks, suitable for accepting an optional first argument with the clojure version to execute tests.
4. Removed the github CI action introduced with the previous commits (i thought there were no tests).
5. Updated the `.circleci` to use the new next generation images (as per https://circleci.com/docs/circleci-images/), download bb and run all tests with clojure 10 and 11

I am happy to remove Planck if you think is now superfluous (I kinda like it, though I never used it :))

borkdude commented on src/respeced/test.cljc:
```diff
@@ -55,18 +55,22 @@
+       ;;
+       ;; Avoid having to write a cross clj[s] pattern for the
+       ;; optional `#'` by using starts/ends-with.
+       (clojure.string/starts-with? msg# "Call to ")
```
@ikappaki Should this be wrapped in `and`?

ikappaki replied:
Well caught! Fixed.

borkdude commented:
Looks good to me, just one remark.

ikappaki commented:
> Looks good to me, just one remark.

I've fixed the exception msg matching and removed Planck as requested.
